plugins/out_s3/s3.c

```diff
@@ -1243,6 +1259,245 @@ static struct multipart_upload *create_upload(struct flb_s3 *ctx,
     return m_upload;
 }
 
+/* Adds an entry to upload queue */
+static void add_to_queue(struct flb_s3 *ctx, struct s3_file *upload_file,
+                 struct multipart_upload *m_upload_file, const char *tag, int tag_len)
+{
+    int ret;
+    struct upload_queue *upload_contents;
+    char *tag_cpy;
+
+    if (upload_file == NULL && m_upload_file == NULL) {
+        flb_plg_error(ctx->ins, "Tried to add entry without file to upload_queue");
+        flb_errno();
+        FLB_OUTPUT_RETURN(FLB_ERROR);
```

Comment:
please avoid using this macro outside of the flush callback. Instead, return some integer (FLB_ERROR or FLB_OK) and use its value with FLB_OUTPUT_RETURN in the code that calls this function/in the flush callback